ecl/hql/hqlgram.y

```diff
@@ -2688,59 +2688,37 @@ actionStmt
                             parser->endList(actions);
                             $$.setExpr(createValue(no_orderedactionlist, makeVoidType(), actions), $1);
                         }
-    | SOAPCALL '(' expression ',' expression ',' recordDef ')'
```

Comment:
I'm not so sure about this change - because it separates the keyword from the '('.  I don't think there are any other examples like it in the parser.

Another alternative might be to have

```
HTTPorSOAPcall
    : HTTPCALL            {   $$.setInt(no_httppost); $$.setPosition($1); }
    | SOAPCALL           {   $$.setInt(no_soapcall); $$.setPosition($1); }
    ;

```
And then
`   node_operator op = ($1.getInt() == no_httppost) ? no_newhttppost : no_newsaopcall;
`

Reply:
There did used to be some code like that for SORT/SORTED.  I'm not sure why it was removed.  Possibly a s/r error when SORTED was added to another production.

Reply:
I also tried that first, and was getting s/r errors. I will create some branches showing my alternatives and maybe get some advice from you on resolving the issues.

Reply:
Btw, I had tried both... with the '(' where you would expect... but that gave errors, and isolating the HTTPCALL and SOAPCALL definition, also errors.  So I'll recreate.